oauth2/build.gradle

```diff
@@ -0,0 +1,34 @@
+ext {
+    immutables_group = 'org.immutables'
+    immutables_version = '2.8.8'
+}
+
+dependencies {
+    api project(':core')
+    api project(':rest')
+
+    compileOnly "$immutables_group:value:$immutables_version"
+    compileOnly "$immutables_group:builder:$immutables_version"
+    annotationProcessor "$immutables_group:value:$immutables_version"
```

Comment:
I was originally going to put those in that repo but thought it would have been misleading since those requests are never serialized as JSON and are instead just serialized directly as strings, but I agree that having a dependency on immutables in this repo seems undesirable. I'll await further feedback to see what everyone thinks.

Reply:
Ran this with the team and though we'd like to use more of immutables in our immutable classes in the future, moving them to Discord4J/discord-json#77 is the right place at this time.

Eventually we might discover more classes in this "immutable but not for serialization" area and could then promote those to a dedicated module